### PR TITLE
Tour start stop

### PR DESCRIPTION
**Pull request by sam33rdhakal**
Updates:
* Starts the tour at the desired tour stop.
* Animates the map to the tour stop's location
* Tour Carousel selects the selected stop as the initial page

### CODE REVIEW

File: map/src/main/kotlin/edu/artic/map/carousel/TourCarouselFragment.kt [hunk 1]
```diff
@@ -114,12 +111,14 @@ class TourCarouselFragment : BaseViewModelFragment<TourCarouselViewModel>() {
...
         viewModel.currentPage
                 .distinctUntilChanged()
+                .observeOn(AndroidSchedulers.mainThread())
                 .subscribe { page ->
                     tourCarousel.setCurrentItem(page, true)
                 }
                 .disposedBy(disposeBag)
 
         tourCarousel.pageSelections()
+                .skip(1)
```
Comment (Cliabhach) [inline map/src/main/kotlin/edu/artic/map/carousel/TourCarouselFragment.kt]: I take it this is intended to skip the first selection of position `0` (the one emitted during layout)?

File: map/src/main/kotlin/edu/artic/map/carousel/TourCarouselViewModel.kt [hunk 2]
```diff
@@ -120,11 +121,10 @@ class TourCarouselViewModel @Inject constructor(private val languageSelector: La
         /**
          * Get index of the object using id and bind it to currentPage.
          */
-        tourProgressManager
-                .selectedStop
-                .withLatestFrom(tourStops) { objectId, tourStops ->
-                    tourStops.map { it.objectId }.indexOf(objectId)
-                }
+        Observables.combineLatest(tourProgressManager.selectedStop, tourStops)
```
Comment (Cliabhach) [inline map/src/main/kotlin/edu/artic/map/carousel/TourCarouselViewModel.kt]: This observes simultaneously, huh. Good catch.

File: tours/src/main/kotlin/edu/artic/tours/TourDetailsViewModel.kt [hunk 3]
```diff
@@ -139,6 +140,14 @@ class TourDetailsViewModel @Inject constructor(
                 }
                 .disposedBy(disposeBag)
     }
+
+    fun stopClicked(viewModel: TourDetailsStopCellViewModel) {
```
[the remark this answers is not in the document]
Comment (sam33rdhakal) [reply tours/src/main/kotlin/edu/artic/tours/TourDetailsViewModel.kt]: Haha, Updated.

File: tours/src/main/kotlin/edu/artic/tours/TourDetailsFragment.kt [hunk 4]
```diff
@@ -117,6 +117,11 @@ class TourDetailsFragment : BaseViewModelFragment<TourDetailsViewModel>() {
                 .bindToMain(adapter.itemChanges())
                 .disposedBy(disposeBag)
 
+        adapter.itemClicks()
+                .subscribeBy {item ->
```
[the remark this answers is not in the document]
Comment (sam33rdhakal) [reply tours/src/main/kotlin/edu/artic/tours/TourDetailsFragment.kt]: Done.

File: map/src/main/kotlin/edu/artic/map/carousel/TourCarouselViewModel.kt [hunk 5]
```diff
@@ -120,11 +121,10 @@ class TourCarouselViewModel @Inject constructor(private val languageSelector: La
...
+        { objectId, tourStops ->
+            tourStops.map { it.objectId }.indexOf(objectId)
```
[the remark this answers is not in the document]
Comment (sam33rdhakal) [reply map/src/main/kotlin/edu/artic/map/carousel/TourCarouselViewModel.kt]: Updated.
Didn't know that method exists. Looks cleaner too. Thanks.

File: map/src/main/kotlin/edu/artic/map/carousel/TourCarouselViewModel.kt [hunk 6]
```diff
@@ -36,7 +37,7 @@ class TourCarouselViewModel @Inject constructor(private val languageSelector: La
     val currentTrack: Subject<Optional<AudioFileModel>> = BehaviorSubject.createDefault(Optional(null))
     val audioPlayBackStatus: Subject<AudioPlayerService.PlayBackState> = BehaviorSubject.create()
     val playerControl: Subject<TourCarousalBaseViewModel.PlayerAction> = PublishSubject.create()
-    val currentPage: Subject<Int> = BehaviorSubject.createDefault(0)
+    val currentPage: Subject<Int> = BehaviorSubject.create()
```
[the remark this answers is not in the document]
Comment (sam33rdhakal) [reply map/src/main/kotlin/edu/artic/map/carousel/TourCarouselViewModel.kt]: We need to persist the current page.

File: map/src/main/kotlin/edu/artic/map/carousel/TourCarouselFragment.kt (hunk 1 above)
Comment (sam33rdhakal) [reply map/src/main/kotlin/edu/artic/map/carousel/TourCarouselFragment.kt]: Replaced `skip` to `skipInitialValue`

File: tours/src/main/kotlin/edu/artic/tours/TourDetailsFragment.kt [hunk 7]
```diff
@@ -85,6 +85,10 @@ class TourDetailsFragment : BaseViewModelFragment<TourDetailsViewModel>() {
                 .bindToMain(tourDetailIntroCell.tourStopTitle.text())
                 .disposedBy(disposeBag)
 
+        tourDetailIntroCell.setOnClickListener {
```
Comment (raigex) [inline tours/src/main/kotlin/edu/artic/tours/TourDetailsFragment.kt]: use `.clicks()` to follow rx

Comment (sam33rdhakal) [reply tours/src/main/kotlin/edu/artic/tours/TourDetailsFragment.kt]: Updated.

**Pull request merged**